mp/src/game/shared/momentum/mom_stickybomb.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "baseprojectile.h"
+
+#ifdef CLIENT_DLL
+#define CMomStickybomb C_MomStickybomb
+#else
+class CMomentumStickybombLauncher;
+#endif
+
+class CMomStickybomb : public CBaseProjectile
+{
+  public:
+    DECLARE_CLASS(CMomStickybomb, CBaseProjectile);
+    DECLARE_NETWORKCLASS();
+
+    // This gets sent to the client and placed in the client's interpolation history
+    // so the projectile starts out moving right off the bat.
+    CNetworkVector(m_vInitialVelocity);
+
+    CMomStickybomb();
+    ~CMomStickybomb();
+
+    float GetCreationTime() { return m_flCreationTime; }
+    void SetChargeTime(float flChargeTime) { m_flChargeTime = flChargeTime; }
+
+    CNetworkHandle(CBaseEntity, m_hLauncher);
```

Comment:
Can this be `CNetworkHandle(CMomentumStickybombLauncher, m_hLauncher)`?

Reply:
Changing the type of this seems to break a lot of stuff and it seems I'd have to refactor a lot of the code to make it work.